internal/xcontext/merge_contexts.go

```diff
@@ -0,0 +1,45 @@
+package xcontext
+
+import (
+	"context"
+	"time"
+)
+
+type MergedContexts struct {
+	context.Context                 //nolint:containedctx
+	BaseContext     context.Context //nolint:containedctx
+	RequestContext  context.Context //nolint:containedctx
+}
+
+func (ctx MergedContexts) Deadline() (deadline time.Time, ok bool) {
+	return ctx.RequestContext.Deadline()
+}
+
+func (ctx MergedContexts) Done() <-chan struct{} {
+	return ctx.RequestContext.Done()
+}
+
+func (ctx MergedContexts) Err() error {
+	return ctx.RequestContext.Err()
+}
+
+func (ctx MergedContexts) Value(key interface{}) interface{} {
+	if ctx.RequestContext.Value(key) != nil {
+		return ctx.RequestContext.Value(key)
+	}
+
+	return ctx.BaseContext.Value(key)
+}
+
+func (ctx MergedContexts) AsCtx() *context.Context {
```

Comment:
Do you really need the method?
may be better some helper function for use with any context?

some similar to:
```
ctxPointer(ctx *context.Context, f func (ctx context.Context)) {
  copyCtx := *ctx
  f(&copyCtx)
  *ctx = copyCtx
})
```
?

Reply:
removed `AsCtx()`, helper function is not really needed because it creates more boilerplate when `trace.DoSmth(ctx, arg1, arg2)` is wrapped in `f` with one argument